src/iexamine.cpp

```diff
@@ -3962,18 +3962,20 @@ void iexamine::autodoc( player &p, const tripoint &examp )
     }
 
     bool needs_anesthesia = true;
-    std::vector<item_comp> acomps;
+    std::vector< tool_comp > anesth_kit;
     if( p.has_trait( trait_NOPAIN ) || p.has_bionic( bionic_id( "bio_painkiller" ) ) ) {
         needs_anesthesia = false;
     } else {
-        std::vector<const item *> a_filter = p.crafting_inventory().items_with( []( const item & it ) {
-            return it.has_flag( "ANESTHESIA" );
+        std::vector< const item *> a_filter = p.crafting_inventory().items_with( []( const item & it ) {
+            return it.has_quality( quality_id( "ANESTHESIA" ) );
         } );
         for( const item *anesthesia_item : a_filter ) {
-            acomps.push_back( item_comp( anesthesia_item->typeId(), 1 ) );
+            if( anesthesia_item->ammo_remaining() > 1 ) {
```

Comment:
Don't *think* you need this check here anymore

Reply:
But if we remove this check where does the "You need _blablabla_ for autodoc to perform any operation." goes ?

Reply:
Oh my bad, you're totally right.